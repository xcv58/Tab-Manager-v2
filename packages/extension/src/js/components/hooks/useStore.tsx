import React from 'react'
import Store from 'stores'

export const store = new Store()

export const StoreContext = React.createContext<Store>(store)

export const useStore = () => {
  return React.useContext(StoreContext)
}

export const useFontSize = () => {
  const { userStore } = useStore()
  return userStore.fontSize
}

export const useTabHeight = () => {
  const fontSize = useFontSize()
  return 3 * fontSize
}
