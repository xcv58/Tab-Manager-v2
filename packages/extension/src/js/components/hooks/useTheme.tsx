import { createContext, useContext } from 'react'

export const ThemeContext = createContext(false)

export const useTheme = () => useContext(ThemeContext)
