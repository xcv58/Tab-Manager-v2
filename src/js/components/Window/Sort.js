import React from 'react'
import { inject, observer } from 'mobx-react'
import Tooltip from 'material-ui/Tooltip'
import SortIcon from '@material-ui/icons/Sort'
import IconButton from 'material-ui/IconButton'

@inject('arrangeStore')
@observer
export default class Sort extends React.Component {
  render () {
    const {
      win: { id },
      arrangeStore: { sortTabs }
    } = this.props
    return (
      <Tooltip title='Sort Tabs'>
        <IconButton onClick={() => sortTabs(id)}>
          <SortIcon />
        </IconButton>
      </Tooltip>
    )
  }
}
