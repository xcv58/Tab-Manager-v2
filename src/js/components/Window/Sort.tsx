import React from 'react'
import { observer } from 'mobx-react'
import Tooltip from '@material-ui/core/Tooltip'
import SortIcon from '@material-ui/icons/Sort'
import IconButton from '@material-ui/core/IconButton'
import { useStore } from 'components/StoreContext'

export default observer(props => {
  const { arrangeStore } = useStore()
  const {
    win: { id }
  } = props
  const { sortTabs } = arrangeStore
  return (
    <Tooltip title='Sort tabs'>
      <IconButton onClick={() => sortTabs(id)}>
        <SortIcon />
      </IconButton>
    </Tooltip>
  )
})
