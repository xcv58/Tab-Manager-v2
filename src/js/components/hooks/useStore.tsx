import React from 'react'
import Store from 'stores'

export const store = new Store()

export const StoreContext = React.createContext<Store>(store)

export const useStore = () => {
  return React.useContext(StoreContext)
}
