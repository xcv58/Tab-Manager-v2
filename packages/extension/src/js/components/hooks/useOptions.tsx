import { useStore } from 'components/hooks/useStore'

export const useOptions = () => {
  const { windowStore, shortcutStore, searchStore } = useStore()
  if (searchStore.isCommand) {
    const { shortcuts } = shortcutStore
    return shortcuts
      .map(([shortcut, command, name, hideFromCommand]) => {
        if (typeof name !== 'string' || hideFromCommand) {
          return null
        }
        return { name, shortcut, command }
      })
      .filter((x) => x)
      .sort((a, b) => a.name.localeCompare(b.name))
  }
  const { historyTabs } = searchStore
  console.log({ historyTabs: [...historyTabs] })
  return [...windowStore.tabs, ...historyTabs]
}
