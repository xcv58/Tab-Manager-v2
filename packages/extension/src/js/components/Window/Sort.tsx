import React from 'react'
import { observer } from 'mobx-react-lite'
import Tooltip from '@material-ui/core/Tooltip'
import SortIcon from '@material-ui/icons/Sort'
import IconButton from '@material-ui/core/IconButton'
import { useStore } from 'components/hooks/useStore'
import { WinProps } from 'components/types'

export default observer((props: WinProps) => {
  const { arrangeStore } = useStore()
  const { id } = props.win
  const { sortTabs } = arrangeStore
  return (
    <Tooltip title='Sort tabs'>
      <IconButton onClick={() => sortTabs(id)} className='focus:outline-none'>
        <SortIcon />
      </IconButton>
    </Tooltip>
  )
})
