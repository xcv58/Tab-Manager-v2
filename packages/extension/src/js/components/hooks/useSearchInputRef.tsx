import { useRef, useEffect } from 'react'
import { useStore } from 'components/hooks/useStore'

export const useSearchInputRef = () => {
  const searchInputRef = useRef<HTMLInputElement>(null)
  const { searchStore } = useStore()
  const { setSearchEl } = searchStore

  useEffect(() => setSearchEl(searchInputRef))
  return searchInputRef
}
