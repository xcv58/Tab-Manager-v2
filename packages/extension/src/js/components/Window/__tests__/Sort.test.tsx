import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import Sort from 'components/Window/Sort'
import * as StoreContext from 'components/hooks/useStore'
import { act } from 'react-dom/test-utils'

const id = 'id'
const sortTabs = jest.fn()
const props = {
  win: { id },
}
const mockStore = {
  arrangeStore: { sortTabs },
}

describe('Sort', () => {
  jest.spyOn(StoreContext, 'useStore').mockImplementation(() => mockStore)

  it('should call arrangeStore.sortTabs', () => {
    render(<Sort {...props} />)
    expect(screen.getByRole('button')).toBeInTheDocument()
    fireEvent.click(screen.getByRole('button'))
    expect(sortTabs.mock.calls.length).toBe(1)
    expect(sortTabs.mock.calls[0]).toEqual([id])
    expect(screen.getByRole('button')).toMatchSnapshot()
  })

  it('should have correct tooltip', async () => {
    const { container } = render(<Sort {...props} />)
    const tooltip = screen.getByRole('button')
    act(() => {
      fireEvent(
        tooltip,
        new MouseEvent('mouseover', {
          bubbles: true,
        })
      )
    })

    // Wait for the tooltip to show up
    const tooltipText = await screen.findByRole('tooltip')
    expect(tooltipText).toBeInTheDocument()
    expect(tooltipText).toMatchSnapshot()
    expect(container).toMatchSnapshot()
  })
})
