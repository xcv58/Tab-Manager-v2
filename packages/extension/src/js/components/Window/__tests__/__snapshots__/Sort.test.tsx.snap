// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort should call arrangeStore.sortTabs 1`] = `
<button
  aria-label="Sort tabs"
  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium focus:outline-none css-78trlr-MuiButtonBase-root-MuiIconButton-root"
  data-mui-internal-clone-element="true"
  tabindex="0"
  type="button"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="SortIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M3 18h6v-2H3v2zM3 6v2h18V6H3zm0 7h12v-2H3v2z"
    />
  </svg>
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`Sort should have correct tooltip 1`] = `
<div
  class="MuiTooltip-popper MuiTooltip-popperInteractive css-yem0l2-MuiPopper-root-MuiTooltip-popper MuiPopperUnstyled-root"
  data-popper-escaped=""
  data-popper-placement="bottom"
  data-popper-reference-hidden=""
  id=":r1:"
  role="tooltip"
  style="position: absolute; top: 0px; left: 0px; margin: 0px; transform: translate(0px, 0px);"
>
  <div
    class="MuiTooltip-tooltip MuiTooltip-tooltipPlacementBottom css-ja5taz-MuiTooltip-tooltip"
    data-foo="bar"
    style="opacity: 1; transform: scale(1, 1); transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 133ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    Sort tabs
  </div>
</div>
`;

exports[`Sort should have correct tooltip 2`] = `
<div>
  <button
    aria-label="Sort tabs"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium focus:outline-none css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    data-mui-internal-clone-element="true"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="SortIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 18h6v-2H3v2zM3 6v2h18V6H3zm0 7h12v-2H3v2z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
